params/config.go

```diff
@@ -905,7 +905,7 @@ func (c *ChainConfig) Rules(num *big.Int, isMerge bool, timestamp *big.Int) Rule
 		IsBerlin:         c.IsBerlin(num),
 		IsLondon:         c.IsLondon(num),
 		IsMerge:          isMerge,
-		IsShanghai:       c.IsShanghai(timestamp),
+		IsShanghai:       timestamp != nil && (!timestamp.IsUint64() || c.IsShanghai(timestamp.Uint64())),
```

Comment:
If timestamp is not Uint64, then IsShanghai is marked as true. It feels weird.

First of all, timestamp bigInt should always be able to be represented as Uint64

```
>>> 2 ** 64 - 1
18446744073709551615
```

If we know it's timestamp and exceed the MAXUINT64, we should just panic because of invalid value.